src/main/java/org/qubership/itool/context/FlowContextImpl.java

```diff
@@ -74,27 +77,39 @@ public class FlowContextImpl implements FlowContext {
     private ClassLoader taskClassLoader;
     private Vertx vertx;
     private boolean breakRequested;
-
-    /* Create a flow context not bound to any graph service and create a new graph in it. */
-    public FlowContextImpl() {
-        this((GraphService)null);
+    private final GraphFactory graphFactory;
+    private final GraphReportFactory graphReportFactory;
+    private final GraphMergerFactory graphMergerFactory;
+
+    @Inject
+    public FlowContextImpl(GraphFactory graphFactory, GraphReportFactory graphReportFactory, GraphMergerFactory graphMergerFactory) {
```

Comment:
Have you considered `FlowContextImpl(Provder<Graph> graphProvider, Provider<GraphReport> graphReportProvider, Provider<GraphMerger> graphMergerProvider) {` instead?

Then the factory interfaces and classes won't be needed.